packages/odata-common/test/common-entity.ts

```diff
@@ -0,0 +1,148 @@
+/* eslint-disable max-classes-per-file */
+import {
+  Constructable,
+  EntityBase,
+  EntityBuilderType,
+  CustomField,
+  FieldBuilder,
+  Field,
+  FieldOptions,
+  EdmTypeField,
+  ComplexTypeField,
+  ConstructorOrField,
+  PropertyMetadata,
+  AllFields
+} from '../src/internal';
+
+export interface CommonEntityComplexType {
+  stringProperty: string;
+  booleanProperty?: boolean;
+}
+export class CommonEntityComplexTypeField<
+  EntityT extends EntityBase
+> extends ComplexTypeField<EntityT, CommonEntityComplexType, true, true> {
+  _fieldBuilder: FieldBuilder<this> = new FieldBuilder(this);
+  stringProperty: EdmTypeField<EntityT, 'Edm.String', false, false> =
+    this._fieldBuilder.buildEdmTypeField('StringProperty', 'Edm.String', false);
+
+  booleanProperty: EdmTypeField<EntityT, 'Edm.Boolean', true, false> =
+    this._fieldBuilder.buildEdmTypeField(
+      'BooleanProperty',
+      'Edm.Boolean',
+      true
+    );
+  constructor(
+    fieldName: string,
+    fieldOf: ConstructorOrField<EntityT>,
+    fieldOptions?: FieldOptions<true, true>
+  ) {
+    super(fieldName, fieldOf, CommonEntityComplexType, fieldOptions);
+  }
+}
+export namespace CommonEntityComplexType {
+  export const _propertyMetadata: PropertyMetadata<CommonEntityComplexType>[] =
+    [
+      {
+        originalName: 'StringProperty',
+        name: 'stringProperty',
+        type: 'Edm.String',
+        isCollection: false
+      },
+      {
+        originalName: 'BooleanProperty',
+        name: 'booleanProperty',
+        type: 'Edm.Boolean',
+        isCollection: false
+      }
+    ];
+}
+
+// eslint-disable-next-line import/export
+export class CommonEntity extends EntityBase {
+  static _entityName = 'A_CommonEntity';
+  static _defaultServicePath = '/sap/opu/odata/sap/API_COMMON_ENTITY_SRV';
+
+  static requestBuilder(): any {
+    throw new Error('not implemented');
+  }
+
+  static customField(): CustomField<CommonEntity> {
+    throw new Error('not implemented');
+  }
+  static builder(): EntityBuilderType<CommonEntity, CommonEntityType> {
+    return EntityBase.entityBuilder(CommonEntity);
+  }
+  readonly _oDataVersion: any;
+  keyPropertyGuid!: string;
+  keyPropertyString!: string;
+  stringProperty: string;
+  int16Property: number;
+}
+export interface CommonEntityType {
+  keyPropertyGuid: string;
+  keyPropertyString: string;
+  stringProperty: string;
+  int16Property: number;
+}
+// eslint-disable-next-line import/export
+export namespace CommonEntity {
```

Comment:
I assume this is not generated, which means we have to remember this test entity when touching the generator.
When Marika's PR gets merged, for example, we might forget this, and therefore it will test the namespace which does not exist.
I feel we might want to use real test service (v2), unless the generator is able to generate a common test entity, which is useless in productive scenario.

Reply:
hm, I guess you are right. I will try to use the generator to get these files.

Reply:
I added generation for the entity and replaced all imports after generation with the `odata-common`